README.md

```diff
@@ -0,0 +1,617 @@
+- Parsed AsyncAPI document that you can access in the filter comes from the AsyncAPI JavaScript Parser. It provides you with extensive [API](https://github.com/asyncapi/parser-js/blob/master/API.md) that you can use to navigate through the document to extract certain parts from it, like for example:
+  ```js
+    const OPERATIONS = ['publish', 'subscribe'];
+
+    asyncapi.channelNames().forEach(channelName => {
+      const channel = asyncapi.channel(channelName);
+
+      OPERATIONS.map((opName) => {
+        const op = channel[opName]();
+        if (!op) return;
+        
+        op.messages().forEach(m => {
+          //logic responsible for processing every message in every operation
+        });
+      });
+    });
+  ```
+- Example filter is used in [template/index.html](template/index.html) using Nunjucks syntax `{{ asyncapi | generateMermaidDiagram }}`
+
+### Hooks
+
+#### Custom Template Hooks
+
+Check out  [hooks/generateExtraFormats.js](hooks/generateExtraFormats.js) file to see an example of the hook. It is a hook that is invoked by the Generator after the template generation process is completed. Generator passes its context to hooks, which means you have access to data like `targetDir` that is a path to the directory where Template is generated, or `templateParams` with information about custom parameters passed by the user. This example hook provides optional features to the Template, like PNG/PDF/SVG generation, that the user decides on with custom parameters. That is not the only use case for the hook. There are more use cases like, for example, template cleanup operations after generation or modifications of the AsyncAPI document right before the generation.
+
+Important things to notice:
+- Hooks, like in the case of the filter, is a regular JavaScript function that can have its dependencies like `const puppeteer = require('puppeteer');` that you need to add to your `package.json` file.
+- Remember to specify what hook type your hook function belongs to
+  ```js
+  module.exports = {
+    'generate:after': yourFunction
+  }
+  ```
+- Generator is a library that also has a CLI. Parameters passed to the Template using CLI are always of String type. This means that even if your parameter accepts Boolean values like true/false, you still need to check them like a String value
+  ```js
+  const parameters = generator.templateParams;
+  if (parameters['pdf'] === 'true') await page.pdf({ format: 'A4', path: `${targetDir}/index.pdf` });
+  ```
+
+#### Official AsyncAPI Hooks
+
+Hooks are reusable between templates. AsyncAPI Initiative provides a library of hooks. You can also create such a library for your templates. You add such a library to `dependencies` in the `package.json` file and configure in `generator.hooks` section like:
```

Comment:
Maybe a link to https://github.com/asyncapi/generator/blob/master/docs/authoring.md#hooks?